packages/pigeon/mock_handler_tester/test/widget_test.dart

```diff
@@ -46,7 +53,34 @@ void main() {
     final Mock mock = Mock();
     TestHostApi.setup(mock);
     final SearchReply reply = await api.search(SearchRequest()..query = 'foo');
-    expect(mock.didCall, true);
+    expect(mock.log, <String>['search']);
     expect(reply.result, 'foo');
   });
+
+  test('no-arg calls', () async {
+    final Api api = Api();
+    final Mock mock = Mock();
+    TestHostApi.setup(mock);
+    await api.initialize();
+    expect(mock.log, <String>['initialize']);
+  });
+
+  test('calling methods with null', () async {
+    final Mock mock = Mock();
+    TestHostApi.setup(mock);
+    expect(await const BasicMessageChannel<Object>(
+      'dev.flutter.pigeon.Api.initialize',
+      StandardMessageCodec(),
+    ).send(null), isEmpty);
+    try {
+      await const BasicMessageChannel<Object>(
+        'dev.flutter.pigeon.Api.search',
+        StandardMessageCodec(),
+      ).send(null) as Map<Object, Object>;
+      expect(true, isFalse); // should not reach here
+    } catch (error) {
+      expect(error, isAssertionError);
```

Comment:
does it make sense to also verify the message?

Reply:
done